File: lib/decidim/cie.rb
# Copyright (C) 2022 Formez PA
# This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License as published by the Free Software Foundation, version 3.
# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
# You should have received a copy of the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>

# frozen_string_literal: true

require "omniauth/strategies/cie_saml"

require_relative "cie/engine"
require_relative "cie/admin"
require_relative "cie/admin_engine"
require_relative "cie/authentication"
require_relative "cie/verification"
require_relative "cie/component"

module Decidim
  module Cie
    autoload :Tenant, "decidim/cie/tenant"

    class << self
      def tenants
        @tenants ||= []
      end

      def test!
        @test = true
      end

      def configure(&block)
        tenant = Decidim::Cie::Tenant.new(&block)
        tenants.each do |existing|
          if tenant.name == existing.name
            raise(
              TenantSameName,
              "Definisci il nome del Tenant. Il nome \"#{tenant.name}\" è già in uso."
            )
          end

          match = tenant.name =~ /^#{existing.name}/
          match ||= existing.name =~ /^#{tenant.name}/
          next unless match

        end

        tenants << tenant
      end

      def setup!
        raise "Il modulo CIE è già stato inizializzato!" if initialized?

        @initialized = true
        tenants.each(&:setup!)
      end

      def find_tenant(name)
        Decidim::Cie.tenants.select { |a| a.name == name}.try(:first)
      end

      private

      def initialized?
        @initialized
      end
    end

    class TenantSameName < StandardError; end

    class InvalidTenantName < StandardError; end
  end
end

File: app/packs/src/decidim/cie/cie.js
// Copyright (C) 2022 Formez PA
// This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License as published by the Free Software Foundation, version 3.
// This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
// You should have received a copy of the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>

'use strict';

var _popupCie = function _popup() {
    var buttons = document.querySelectorAll('.cie-italia-it-button');
    var popupContainer = document.querySelector('.cie-container');

    if (buttons.length > 0) {
        (function () {

            popupContainer.querySelectorAll( "[data-idp] img").forEach(function (button) {
                button.addEventListener('click', function (e) {
                    e.stopPropagation();
                    button.closest( "form").submit();
                });
            });
        })();
    }
};

document.addEventListener("DOMContentLoaded", function (event) {
    _popupCie();
});

document.addEventListener('turbolinks:load', function () {
    _popupCie();
});